packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmAnyTests.kt

```diff
@@ -421,6 +422,66 @@ class RealmAnyTests {
         }
     }
 
+    @Test
+    fun equals() {
+        RealmAny.Type.values().forEach { type ->
+            when (type) {
+                RealmAny.Type.INT -> {
+                    assertEquals(RealmAny.create(1), RealmAny.create(1))
+                    assertNotEquals(RealmAny.create(1), RealmAny.create(2))
+                }
+                RealmAny.Type.BOOL -> {
+                    assertEquals(RealmAny.create(true), RealmAny.create(true))
+                    assertNotEquals(RealmAny.create(true), RealmAny.create(false))
+                }
+                RealmAny.Type.STRING -> {
+                    assertEquals(RealmAny.create("Realm"), RealmAny.create("Realm"))
+                    assertNotEquals(RealmAny.create("Realm"), RealmAny.create("Not Realm"))
+                }
+                RealmAny.Type.BINARY -> {
+                    assertEquals(
+                        RealmAny.create(byteArrayOf(1, 2)), RealmAny.create(byteArrayOf(1, 2))
+                    )
+                    assertNotEquals(
+                        RealmAny.create(byteArrayOf(1, 2)), RealmAny.create(byteArrayOf(2, 1))
+                    )
+                }
+                RealmAny.Type.TIMESTAMP -> {
+                    val now = RealmInstant.now()
+                    assertEquals(RealmAny.create(now), RealmAny.create(now))
+                    assertNotEquals(RealmAny.create(RealmInstant.from(1, 1)), RealmAny.create(now))
+                }
+                RealmAny.Type.FLOAT -> {
```

Comment:
The values here should be something that cannot be implicitly cast to a Long

Reply:
The `toLong()` actually truncates it, so can't really choose other good numbers. I removed the `toLong()` implementation in `RealmAny.equals` and are now relying on inner value's `equals` so should behave according to Kotlins semantics now.